main.py

```diff
@@ -41,16 +51,69 @@ def set_img_cnt(cnt: int):
     with open(FILEPATH + "/img_cnt.txt", "w", encoding="utf-8") as file:
         file.write(str(cnt))
 
+# https://github.com/jdemaeyer/ice/blob/master/ice/__init__.py
+def set_config(camera, context, config_name, value):
+    """
+    Change a setting on the camera
+    """
+    log("Setting '{}' to '{}'".format(config_name, value))
+    config = gp.check_result(
+                gp.gp_camera_get_config(camera, context))
+    widget = gp.check_result(
+                gp.gp_widget_get_child_by_name(config, config_name))
+    gp.check_result(gp.gp_widget_set_value(widget, value))
+    gp.check_result(
+            gp.gp_camera_set_config(camera, config, context))
+    log("Set '{}' to '{}'".format(config_name, value))
 
 def take_image():
-    if stop():
-        log('Detected content in "stop.txt" file. Images will no longer be taken.')
-        sys.exit()
-    with lock:
-        threading.Timer(2.0, take_image).start()
-        last_image = get_img_cnt()
-        set_img_cnt(last_image + 1)
-        subprocess.Popen(IMAGE_COMMAND + [f"{FILEPATH}/assets/images/{last_image + 1}.png"])
+    log('Please connect and switch on the camera')
+    error, camera = gp.gp_camera_new()
+    while True: # Wait for camera to be connected
+        error = gp.gp_camera_init(camera)
+        if error >= gp.GP_OK:
+            # operation completed successfully so exit loop
+            log("Camera Connected")
+            break
+        if error != gp.GP_ERROR_MODEL_NOT_FOUND:
+            # some other error we can't handle here
+            raise gp.GPhoto2Error(error)
+        log("No Camera Found, trying again")
+        time.sleep(2)
+    # Log the detected camera
+    error, text = gp.gp_camera_get_summary(camera)
+    log(text.text)
+    context = gp.gp_context_new()
+    # Set aperture, iso, shutterspeed
+    # The camera takes some time to "ramp up" to the setting instead of instantly setting the setting, so we wait 2 seconds between each setting change
+    time.sleep(2)
+    set_config(camera, context, "f-number", config["image"]["f-number"])
+    time.sleep(2)
+    set_config(camera, context, "iso", config["image"]["iso"])
+    time.sleep(2)
+    set_config(camera, context, "shutterspeed", config["image"]["shutterspeed"])
+    captime = time.perf_counter()
```

Comment:
Why `time.perf_counter()` instead of `time.time()`? We want to use system-wide time, not CPU time, right?

> On modern systems, [perf_counter] is measured using a CPU counter which is monotonically increased at a frequency related to CPU's hardware clock. The counter resolution is highly dependent on the system's hardware, the value cannot be reliably related to real-world time or even compared between systems in most cases. Furthermore, the counter value is reset every time the CPU is powered up or reset.